Tools/autotest/pysim/util.py

```diff
@@ -105,35 +120,38 @@ def pexpect_drain(p):
     except pexpect.TIMEOUT:
         pass
 
-def start_SIL(atype, valgrind=False, gdb=False, wipe=False, synthetic_clock=True, home=None, model=None, speedup=1, defaults_file=None):
+def cmd_as_shell(cmd):
+    return (" ".join([ '"%s"' % x for x in cmd ]))
+
+def start_SIL(binary, valgrind=False, gdb=False, wipe=False, synthetic_clock=True, home=None, model=None, speedup=1, defaults_file=None):
     '''launch a SIL instance'''
     import pexpect
-    cmd=""
+    cmd=[]
     if valgrind and os.path.exists('/usr/bin/valgrind'):
-        cmd += 'valgrind -q --log-file=%s-valgrind.log ' % atype
+        cmd.extend(['valgrind', '-q', '--log-file', '%s-valgrind.log' % atype])
```

Comment:
On Thu, 26 May 2016, Francisco Ferreira wrote:

> 
> In Tools/autotest/pysim/util.py:
> 
> >      if valgrind and os.path.exists('/usr/bin/valgrind'):
> > -        cmd += 'valgrind -q --log-file=%s-valgrind.log ' % atype
> > +        cmd.extend(['valgrind', '-q', '--log-file', '%s-valgrind.log' % atype])
> 
> You have atype here but you don't receive it anymore.

Darn.  Will I make "binary" shell-safe and use that instead?  I wonder if 
anything relies on that filename...